tests/client/net_handler_client.c

```diff
@@ -307,6 +307,12 @@ void start_net_handler() {
             printf("pthread_create: output dump\n");
         }
         usleep(400000);  // allow time for thread to dump output before returning to client
+        
+        // execute the keyboard_interface on a seperate thread
+        pthread_t keyboard_id; // id of thread running the keyboard_interface
+        if(pthread_create(&keyboard_id, NULL, (void *)setup_keyboard, NULL) != 0) {
+            printf("pthread create: setup keyboard");
```

Comment:
Did you test that this works? Maybe I'm dumb but I can't seem to find the function `setup_keyboard` that is supposed to be running in this thread

Reply:
The function is implemented in `keyboard_interface.c`, but declared in `net_handler_client.h`. I think it's just a mistake :P I made a comment about this in my review below: maybe the "keyboard interface" functions can be merged with net handler client?

Reply:
Yea so I did test it and it was on purpose. So I had thought about moving the functions over to net handler client but kind of wanted to leave it in its own file since it does its own communicating with the python script and processing that the `net_handler_client` doesn't do. I also wasn't the biggest fan of making a `.h` for like one function that would be in it. So in the Makefile I added the keyboard_interface under the net_handler_cli's rules:
`net_handler_cli: cli/net_handler_cli.* client/net_handler_client.* cli/keyboard_interface.* ../net_handler/net_util.* \`

Reply:
I might just end up putting it in its own `.h` right now though to clean up the `keyboard_interface` file since it also has enums and some include statements in there

Reply:
I haven't seen functions declared like this before--I would have made a header file for just that one function. Maybe Ben can comment on this because he has that K&R experience

Reply:
I think with this change, the intention was for the keyboard interface to be merged with the CLI right?
You're right, in my experience, a function should always be declared in a header file with the same way as the source file in which it is defined. So we should definitely change this as the way it is.
Functionally, I think the keyboard interface stuff is more of a CLI thing (since it's used for parsing input from the command-line, kinda) which it then passes on to the `net_handler` process through `net_handler_client`. So in my view, you're right, the `net_handler_client` probably shouldn't be worried about parsing command-line input. Instead, I think the keyboard logic should be part of the `net_handler_cli`, maybe running as a thread in there

Reply:
Makes sense to me
So we wouldn't have a `keyboard_interface.c` anymore, but rather it would be a part of `net_handler_cli`?

Reply:
Cant we just have the `keyboard_interface` have its own header then and have the cli include that header and just crate a thread in the cli to execute the keyboard_interface?

Reply:
Sure, we could do that too!